logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/ConversationEventReceiver.kt

```diff
@@ -77,7 +72,20 @@ class ConversationEventReceiver(
                     )
                     kaliumLogger.i(message = "Message received: $message")
                     when (message.content) {
-                        is MessageContent.Text, is MessageContent.Asset -> messageRepository.persistMessage(message)
+                        is MessageContent.Text -> messageRepository.persistMessage(message)
+                        is MessageContent.Asset -> {
+                            messageRepository.getMessageById(message.conversationId, message.id)
+                                .onFailure {
+                                    // No asset message was received previously, so just persist the preview asset message
+                                    messageRepository.persistMessage(message)
+                                }
+                                .onSuccess { persistedMessage ->
+                                    // The asset message received contains the asset decryption keys, so update the preview message persisted previously
+                                    updateMessage(persistedMessage, message.content.value.remoteData)?.let {
```

Comment:
We also need to verify that sender of the updated message is the same as the original message.

Reply:
Would it be possible to receive 2 messages with same message id but different sender ID's? If so that would be indicative of something more serious happening

Reply:
It shouldn't be possible for someone else then the original sender to update his/her message. If we don't verify this we have security vulnerability since then an attacker could secretly replace the asset sent by someone else.